lib/mumukit/templates/multi_file_hook.rb

```diff
@@ -0,0 +1,25 @@
+module Mumukit
+  class Templates::MultiFileHook < Templates::FileHook
+    include Mumukit::Templates::WithMultipleFiles
+
+    def compile(request)
+      return super(request) unless has_files?(request)
+
+      self.request = request
+      compile_file_content(request).map { |filename, content| write_tempfile!(content, filename) }
+    end
+
+    def run!(files)
+      return super(files) unless has_files?(request)
+
+      result, status = run_files!(*files)
+      post_process_file(files, cleanup_raw_result(result), status)
+    ensure
+      if has_files?(request)
+        files.each { |file| file.unlink }
+      else
+        files.unlink
```

Comment:
This is weird. Isn't `files` always a list? Or it may be a single file or a list of files?

Event in the second scenario  we could avoid the if with something like `[files].flatten.each ..`. instead of doing the if.

Reply:
Yes, this hook is backwards compatible with the old single file requests, so it can be a list of files or a single file.

> Event in the second scenario we could avoid the if with something like [files].flatten.each ... instead of doing the if.

Good point!
I think the if was there since I was doing different things (regular files didn't have the `unlink` method until I created the `NamedTempfile`)